site/src/sphinx/server-annotated-service.rst

```diff
@@ -0,0 +1,627 @@
+            // 'there' is converted in a case-insensitive way.
+        }
+
+        @Get("/hello2/{there}")
+        public HttpResponse hello2(@Param("there") CaseSensitive there) {
+            // 'there' must be converted in a case-sensitive way.
+            // So 'ALPHA' and 'alpha' are only acceptable.
+        }
+    }
+
+Handling exceptions
+-------------------
+
+An exception may be raised from a service method, of course. *Armeria* provides `@ExceptionHandler`_
+annotation to transform an exception into a response. You can write your own exception handler
+by implementing ExceptionHandlerFunction_ interface. Then, annotate your service object or method
+with `@ExceptionHandler`_ annotation. Here is an example of an exception handler. If your exception
+handler is not able to handle an exception given, you can call ``ExceptionHandlerFunction.fallThrough()``
+to pass the exception to the next exception handlers.
+
+.. code-block:: java
+
+    class MyExceptionHandler implements ExceptionHandlerFunction {
+        @Override
+        public HttpResponse handleException(RequestContext ctx, HttpRequest req, Throwable cause) {
+            if (cause instanceof MyServiceException) {
+                return HttpResponse.of(HttpStatus.CONFLICT);
+            }
+
+            // To the next exception handler.
+            return ExceptionHandlerFunction.fallthrough();
+        }
+    }
+
+To catch an exception from every method in your service object, annotate your class with the annotation.
+
+.. code-block:: java
+
+    @ExceptionHandler(MyExceptionHandler.class)
+    class MyAnnotatedServiceClass {
+
+        @Get("/hello")
+        public HttpResponse hello() {
+            ...
+        }
+    }
+
+To catch an exception from a single method in your service object, annotate your method with the annotation.
+
+.. code-block:: java
+
+    class MyAnnotatedServiceClass {
+
+        @Get("/hello")
+        @ExceptionHandler(MyExceptionHandler.class)
+        public HttpResponse hello() {
+            ...
+        }
+    }
+
+If there is no exception handler which is able to handle an exception, the exception would be passed to
+the default exception handler. It handles ``IllegalArgumentException``, HttpStatusException_ and
+HttpResponseException_ by default. ``IllegalArgumentException`` would be converted into ``400 Bad Request``
+response, and the other two exceptions would be converted into a response with the status code which
+they are holding. For another exceptions, ``500 Internal Server Error`` would be sent to the client.
+
+Converting between an HTTP message and a Java object
+----------------------------------------------------
+
+Converting an HTTP request to a Java object
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In some cases like receiving a JSON document from a client, it may be useful to convert the document to
+a Java object automatically. So *Armeria* supports that using `@RequestConverter`_ and `@RequestObject`_
```

Comment:
So Armeria Supports that using -> Armeria provides ... annotations so that such conversion can be done conveniently.